pymodbus/client/serial.py

```diff
@@ -14,8 +13,13 @@
 from pymodbus.utilities import ModbusTransactionState
 
 
-with suppress(ImportError):
+try:
     import serial
+except ImportError:
```

Comment:
This is wrong, because it demands users to have the dependency installed.

The existing version, allows the user not to have the dependency installed, but of course he/she is then not allowed to use these methods.

We do NOT want to demand users to install e.g. pyserial / aiohttp if they are going to use serial communication / the simulator.

Reply:
what I did earlier to avoid followup errors was to have

```
except ImportError:
    serial = None
```

Reply:
> We do NOT want to demand users to install e.g. pyserial / aiohttp if they are going to use serial communication / the simulator.

I'm confused.  What use is importing the serial client without `pyserial`, or importing the HTTP simulator without `aiohttp`?  They will not run, and fail with some confusing error message.

Are you concerned about importing `pymodbus` to use another part? (for instance, the TCP client as in https://github.com/pymodbus-dev/pymodbus/issues/1128) I checked and this is not a problem.  
```
>>> import serial
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'serial'
>>> from pymodbus.client import AsyncModbusTcpClient
>>>
```

Last year pymodbus did something like `import *`, but now after your refactors it does not.  So I'm not sure how the ImportError would be triggered when it shouldn't be.

Reply:
Because the user does "import pymodbus" and that scans all py files, and it then breaks down.

Try to uninstall pyserial and try for yourself, I had the problem about a month ago. aiohttp was never the problem, but pyserial have caused a couple of issues.

See this:
```
(pymodbus) jan@MacMini examples % pip uninstall pyserial
...
(pymodbus) jan@MacMini examples % ./client_async.py
Traceback (most recent call last):
  File "/Volumes/DiskHomes/home_jan/repos/pymodbus/pymodbus/client/serial.py", line 17, in <module>
    import serial
ModuleNotFoundError: No module named 'serial'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Volumes/DiskHomes/home_jan/repos/pymodbus/examples/./client_async.py", line 35, in <module>
    from pymodbus.client import (
  File "/Volumes/DiskHomes/home_jan/repos/pymodbus/pymodbus/client/__init__.py", line 16, in <module>
    from pymodbus.client.serial import AsyncModbusSerialClient, ModbusSerialClient
  File "/Volumes/DiskHomes/home_jan/repos/pymodbus/pymodbus/client/serial.py", line 19, in <module>
    raise ImportError(  # pylint: disable=raise-missing-from
ImportError: Serial client requires pyserial Please install with "pip install pyserial" and try again.
(pymodbus) jan@MacMini examples % 
```
client_async without parameters uses a tcp client.

Reply:
Done with this PR.

Reply:
> Because the user does "import pymodbus" and that scans all py files, and it then breaks down. [...]
> Try to uninstall pyserial and try for yourself, I had the problem about a month ago. 

Yes, I did :). It's fine
<img width="375" alt="Screenshot 2023-10-17 at 11 27 37 AM" src="https://github.com/pymodbus-dev/pymodbus/assets/52292902/964d5b71-dbd5-4487-8a7d-04843522aa42">

```
(pymodbus) jan@MacMini examples % ./client_async.py
Traceback (most recent call last):
  File "/Volumes/DiskHomes/home_jan/repos/pymodbus/pymodbus/client/serial.py", line 17, in <module>
    import serial
ModuleNotFoundError: No module named 'serial'
```

OK, now I understand what I missed.  _The examples_ fail, not all of pymodbus.  I'll look to see if that can be changed.